chemist-aws/src/main/scala/AwsDiscovery.scala

```diff
@@ -157,54 +166,51 @@ class AwsDiscovery(
+      ids.map(
+        id => id -> cache.get(id)
+      ).foldLeft[(Seq[String], Seq[AwsInstance])]((Seq.empty, Seq.empty)) {
+        (a,b) =>
           val (ids,instances) = a
           b match {
             case (id,Some(instance)) => (ids, instances :+ instance)
             case (id,None) => (ids :+ id, instances)
           }
-        }
+      }
 
     def lookInAws(specificIds: Seq[String]): Task[Seq[AwsInstance]] =
-      LookupManyAws.timeTask {
-        for {
-          a <- EC2.reservations(specificIds)(ec2)
-          _  = log.debug(s"lookupMany, reservations = ${a.length}")
-          b <- Task.now(a.flatMap(_.getInstances.asScala.map(fromAWSInstance)))
-          (fails,successes) = b.toVector.separate
-          _  = log.info(s"lookupMany, failed to validate ${fails.length} instances.")
-          _  = log.debug(s"lookupMany, validated ${successes.length} instances.")
-          _  = fails.foreach(x => log.error(s"lookupMany, failed to validate '$x'"))
-        } yield successes
-      }
+      if (specificIds.isEmpty)
+        Task.now(Seq.empty)
+      else
+        LookupManyAws.timeTask {
+          for {
+            a <- EC2.reservations(specificIds)(ec2)
+            b <- Task.now(a.flatMap(_.getInstances.asScala.map(fromAWSInstance)))
+            (ignored,eligible) = b.toVector.separate
+            _  = log.info(s"[lookupMany] reservations=${a.length} eligible=${eligible.length} ignored=${ignored.length}")
+            _  = ignored.groupBy(_._2).mapValues(_.map(_._1)).foreach {
+              case (reason, lst) => log.warn(s"[lookupMany] ignored ${lst.size} instances, reason='$reason', ids=$lst")
+            }
+          } yield eligible
+        }.handleWith {
+          case t =>
+            LookupAwsFailure.increment
+            Task.fail(t)
+        }
 
     def updateCache(instances: Seq[AwsInstance]): Task[Seq[AwsInstance]] =
       Task.delay {
-        log.debug(s"lookupMany, updating the cache with ${instances.length} items.")
+        log.debug(s"[lookupMany] updating the cache, items=${instances.length}.")
         instances.foreach(i => cache.put(i.id, i))
         instances
       }
 
-    lookInCache match {
-      // all found in cache
-      case (Nil,found) =>
-        log.debug(s"lookupMany, all ${found.length} instances in the cache.")
-        Task.now(found)
-
-      // none found in cache
-      case (missing,Nil) =>
-        log.debug(s"lookupMany, all ${missing.length} instances are missing in the cache.")
-        lookInAws(missing).flatMap(updateCache)
-
-      // partially found in cache
-      case (missing,found) =>
-        log.debug(s"lookupMany, ${missing.length} missing. ${found.length} found in the cache.")
-        lookInAws(missing)
-          .flatMap(updateCache)
-          .map(_ ++ found)
-    }
+    val (missing, found) = lookInCache
+
+    if (missing.nonEmpty)
+      log.info(s"[lookupMany] missing=${missing.length}, cached=${found.length}")
```

Comment:
i do not follow. What check?

Here, i want to avoid log messages once everything is in the cache. This is 99% of time and it does not help. I am only interested in seeing details about "new" hosts in the log at the `info` level.